pkg/model/api/v1/dashboard.go

```diff
@@ -142,30 +138,21 @@ func (d *Dashboard) validate() error {
 // verifyAndSetJSONRef will check that each JSON Reference are pointing to an existing object and will set the related pointer in the JSONRef.Object
 func (d *Dashboard) verifyAndSetJSONReferences() error {
 	for _, layout := range d.Spec.Layouts {
-		switch parameter := layout.Parameter.(type) {
-		case *dashboard.GridLayoutParameter:
-			for _, line := range parameter.Children {
-				for _, cell := range line {
-					if cell.Content != nil {
-						if err := d.checkAndSetRef(cell.Content); err != nil {
-							return err
-						}
-					}
-				}
-			}
-		case *dashboard.ExpandLayoutParameter:
-			for _, ref := range parameter.Children {
-				if err := d.checkAndSetRef(ref); err != nil {
+		switch spec := layout.Spec.(type) {
+		case *dashboard.GridLayoutSpec:
+			for _, item := range spec.Items {
+				if err := d.checkAndSetRef(item.Content); err != nil {
```

Comment:
I'm curious long-term what validation lives on the FE vs. BE--for instance, does any of the current FE validation need to be updated as well like this [resolvePanelRef function](https://github.com/perses/perses/blob/v0.4.2/ui/core/src/utils/panel-refs.ts#L24) or does that remain the same?

Reply:
The minimum required is that the BE contains all necessary validation so it can ensure that the data are valid and can be read and used by the FE.

Then for reactivity and nice user experience, we could have the same amount of validation on the FE.
Looking at the function you pointed @sjcobb I think we are good. I think this question should be raised again when the UI will be able to create / edit a dashboard. Because then this kind of validation will be necessary to validate the form before sending it to the BE